**РАСЧЕТНО ГРАФИЧЕСКАЯ РАБОТА №2**

Вариант 18

Выполнил:

Суханкулиев Мухаммет,
студент группы N3246

In [290]:
import numpy as np

Значения X и Y

Исходная таблица вероятностей с NULL-значением


In [291]:
X_values = np.array([-9, -4, 6, 20, 26])
Y_values = np.array([-6, 3, 7, 10])

probs = np.array(
    [
        [0.09, 0.09, 0.08, 0.01],
        [0.09, 0.05, 0.08, 0.02],
        [0.08, 0.08, 0.07, 0.06],
        [0.01, 0.01, 0.05, 0.02],
        [0.01, np.nan, 0.01, 0.09],
    ]
)

Проверим условие:
$$\sum_{i} \sum_{j} P_{i,j} = 1$$

In [292]:
sum = np.nansum(probs)
sum

1.0000000000000002

Значит пропущенное значение NULL равно 0. Для корректного последующего выполнения кода определим это:

In [293]:
missing_value = 1 - np.nansum(probs)
probs[4][1] = round(float(missing_value), 4)
probs

array([[ 0.09,  0.09,  0.08,  0.01],
       [ 0.09,  0.05,  0.08,  0.02],
       [ 0.08,  0.08,  0.07,  0.06],
       [ 0.01,  0.01,  0.05,  0.02],
       [ 0.01, -0.  ,  0.01,  0.09]])

Вычислим маргинальные значения для X и Y:
$$P(X = x_i) = \sum_{j} P_{i,j}$$
$$P(Y = y_j) = \sum_{i} P_{i,j}$$

In [294]:
marginal_X = probs.sum(axis=1)
marginal_X

array([0.27, 0.24, 0.29, 0.09, 0.11])

In [295]:
marginal_Y = probs.sum(axis=0)
marginal_Y

array([0.28, 0.23, 0.29, 0.2 ])

Математические ожидание $\mathbb{E}[X]$ и $\mathbb{E}[Y]$:
$$\mathbb{E}[X] = \sum_{i} x_i \cdot P(X = x_i)$$
$$\mathbb{E}[Y] = \sum_{j} y_j \cdot P(Y = y_j)$$
Дисперсии $\operatorname{D}(X)$ и $\operatorname{D}(Y)$:
$$\operatorname{D}(X) = \sum_{i} (x_i - \mathbb{E}[X])^2 \cdot P(X = x_i)$$
$$\operatorname{D}(Y) = \sum_{j} (y_j - \mathbb{E}[Y])^2 \cdot P(Y = y_j)$$
Среднеквадратичные отклонения \( X \) и \( Y \):
$$\sigma_X = \sqrt{\operatorname{D}(X)}$$
$$\sigma_Y = \sqrt{\operatorname{D}(Y)}$$

In [296]:
E_X = np.sum(X_values * marginal_X)
E_X

3.0100000000000002

In [297]:
E_Y = np.sum(Y_values * marginal_Y)
E_Y

3.04

In [298]:
D_X = np.sum((X_values - E_X) ** 2 * marginal_X)
D_X

137.44989999999999

In [299]:
D_Y = np.sum((Y_values - E_Y) ** 2 * marginal_Y)
D_Y

37.1184

In [300]:
sigma_X = np.sqrt(D_X)
sigma_X

11.723902933750347

In [301]:
sigma_Y = np.sqrt(D_Y)
sigma_Y

6.092487176843297

Ковариация $\operatorname{Cov}(X, Y)$ определяется как:
$$\operatorname{Cov}(X, Y) = \sum_{i} \sum_{j} (x_i - \mathbb{E}[X])(y_j - \mathbb{E}[Y]) \cdot P_{i,j}$$
Коэффициент корреляции $\rho(X, Y)$:
$$\rho(X, Y) = \frac{\operatorname{Cov}(X, Y)}{\sigma_X \cdot \sigma_Y}$$

In [302]:
covariance = np.sum((X_values[:, None] - E_X) * (Y_values - E_Y) * probs)
covariance

25.019599999999997

In [303]:
correlation = covariance / (sigma_X * sigma_Y)
correlation

0.350278534684316

Случайные величины \( X \) и \( Y \) независимы, если:
$$P(X = x_i, Y = y_j) = P(X = x_i) \cdot P(Y = y_j) \quad \forall \, i, j$$

In [304]:
independence_check = np.allclose(probs, np.outer(marginal_X, marginal_Y))
independence_check

False

Это значит, что X и Y не независимые

Условные вероятности:
$$P(X = x_i \mid Y = y_j) = \frac{P(X = x_i, Y = y_j)}{P(Y = y_j)}, \quad \text{если } P(Y = y_j) > 0$$
$$P(Y = y_j \mid X = x_i) = \frac{P(X = x_i, Y = y_j)}{P(X = x_i)}, \quad \text{если } P(X = x_i) > 0$$

In [305]:
conditional_X_given_Y = probs / marginal_Y
conditional_X_given_Y

array([[ 0.32142857,  0.39130435,  0.27586207,  0.05      ],
       [ 0.32142857,  0.2173913 ,  0.27586207,  0.1       ],
       [ 0.28571429,  0.34782609,  0.24137931,  0.3       ],
       [ 0.03571429,  0.04347826,  0.17241379,  0.1       ],
       [ 0.03571429, -0.        ,  0.03448276,  0.45      ]])

In [306]:
conditional_Y_given_X = probs.T / marginal_X
conditional_Y_given_X

array([[ 0.33333333,  0.375     ,  0.27586207,  0.11111111,  0.09090909],
       [ 0.33333333,  0.20833333,  0.27586207,  0.11111111, -0.        ],
       [ 0.2962963 ,  0.33333333,  0.24137931,  0.55555556,  0.09090909],
       [ 0.03703704,  0.08333333,  0.20689655,  0.22222222,  0.81818182]])

Условное математическое ожидание \( X \) при условии \( Y = y_j \):
$$\mathbb{E}[X \mid Y = y_j] = \sum_{i} x_i \cdot P(X = x_i \mid Y = y_j)$$

Условное математическое ожидание \( Y \) при условии \( X = x_i \):
$$\mathbb{E}[Y \mid X = x_i] = \sum_{j} y_j \cdot P(Y = y_j \mid X = x_i)$$

In [307]:
conditional_E_X_given_Y = np.sum(X_values[:, None] * conditional_X_given_Y, axis=0)
conditional_E_X_given_Y

array([-0.82142857, -1.43478261,  2.20689655, 14.65      ])

In [308]:
conditional_E_Y_given_X = np.sum(Y_values[:, None] * conditional_Y_given_X, axis=0)
conditional_E_Y_given_X

array([1.44444444, 1.54166667, 2.93103448, 5.77777778, 8.27272727])

In [309]:
# Вывод результатов
print(f"1. Пропущенное значение вероятности: {missing_value:.1f}")
print(f"2. Маргинальные распределения X: {marginal_X}")
print(f"   Маргинальные распределения Y: {marginal_Y}")
print(f"3. Математическое ожидание X: {E_X:.2f}, Y: {E_Y:.2f}")
print(f"   Дисперсия X: {D_X:.4f}, Y: {D_Y:.4f}")
print(f"   Среднеквадратичное отклонение X: {sigma_X:.4f}, Y: {sigma_Y:.4f}")
print(f"4. Ковариация: {covariance:.4f}")
print(f"   Коэффициент корреляции: {correlation:.4f}")
print(f"5. Независимость X и Y: {'Независимы' if independence_check else 'Зависимы'}")
print(f"6. Условные распределения X|Y=y_j: \n {np.round(conditional_X_given_Y, 4)}")
print(f"   Условные распределения Y|X=x_i: \n {np.round(conditional_Y_given_X, 4)}")
print(
    f"7. Условные математические ожидания X|Y=y_j: {np.round(conditional_E_X_given_Y, 4)}"
)
print(
    f"   Условные математические ожидания Y|X=x_i: {np.round(conditional_E_Y_given_X, 4)}"
)

1. Пропущенное значение вероятности: -0.0
2. Маргинальные распределения X: [0.27 0.24 0.29 0.09 0.11]
   Маргинальные распределения Y: [0.28 0.23 0.29 0.2 ]
3. Математическое ожидание X: 3.01, Y: 3.04
   Дисперсия X: 137.4499, Y: 37.1184
   Среднеквадратичное отклонение X: 11.7239, Y: 6.0925
4. Ковариация: 25.0196
   Коэффициент корреляции: 0.3503
5. Независимость X и Y: Зависимы
6. Условные распределения X|Y=y_j: 
 [[ 0.3214  0.3913  0.2759  0.05  ]
 [ 0.3214  0.2174  0.2759  0.1   ]
 [ 0.2857  0.3478  0.2414  0.3   ]
 [ 0.0357  0.0435  0.1724  0.1   ]
 [ 0.0357 -0.      0.0345  0.45  ]]
   Условные распределения Y|X=x_i: 
 [[ 0.3333  0.375   0.2759  0.1111  0.0909]
 [ 0.3333  0.2083  0.2759  0.1111 -0.    ]
 [ 0.2963  0.3333  0.2414  0.5556  0.0909]
 [ 0.037   0.0833  0.2069  0.2222  0.8182]]
7. Условные математические ожидания X|Y=y_j: [-0.8214 -1.4348  2.2069 14.65  ]
   Условные математические ожидания Y|X=x_i: [1.4444 1.5417 2.931  5.7778 8.2727]


**Ответы**
1. Пропущенное значение вероятности: $P_{2,5} = 0$

2. Маргинальные распределения:
$$P(X = x_1) = 0.27, \quad P(X = x_2) = 0.24, \quad P(X = x_3) = 0.29, \quad P(X = x_4) = 0.09, \quad P(X = x_5) = 0.11$$
$$P(Y = y_1) = 0.28, \quad P(Y = y_2) = 0.23, \quad P(Y = y_3) = 0.29, \quad P(Y = y_4) = 0.2$$

3. Математическое ожидание:
$$E[X] = 3.01 \quad E[Y] = 3.04$$
Дисперсия:
$$D[X] = 137.4499 \quad D[Y] = 37.1184$$
Среднеквадратичное отклонение:
$$\sigma_X = 11.7239 \quad \sigma_Y = 6.0925$$

4. Ковариация:
$$\operatorname{Cov}(X, Y) = 25.0196$$
Коэффициент корреляции:
$$\rho(X, Y) = 0.3503$$

5. Независимость случайных величин \( X \) и \( Y \):
$$\text{X и Y} \quad \text{Зависимы}$$

6. Условные распределения:
$$P(X = x_i \mid Y = y_j) =
\begin{bmatrix}
0.3214 & 0.3913 & 0.2759 & 0.05  \\
0.3214 & 0.2174 & 0.2759 & 0.1  \\
0.2857 & 0.3478 & 0.2414 & 0.3  \\
0.0357 & 0.0435 & 0.1724 & 0.1  \\
0.0357 & 0 & 0.0345 & 0.45
\end{bmatrix}$$
$$P(Y = y_j \mid X = x_i) =
\begin{bmatrix}
0.3333 & 0.375 & 0.2759 & 0.1111 & 0.0909 \\
0.3333 & 0.2083 & 0.2759 & 0.1111 & 0 \\
0.2963 & 0.3333 & 0.2414 & 0.5556 & 0.0909 \\
0.037 & 0.0833 & 0.2069 & 0.2222 & 0.8182 \\
\end{bmatrix}$$
7. Условные математические ожидания:
$$E[X \mid Y = y_1] = -0.8214, \quad E[X \mid Y = y_2] = -1.4348, \quad E[X \mid Y = y_3] = 2.2069, \quad E[X \mid Y = y_4] = 14.65$$
$$E[Y \mid X = x_1] = 1.4444, \quad E[Y \mid X = x_2] = 1.5417, \quad E[Y \mid X = x_3] = 2.931, \quad E[Y \mid X = x_4] = 5.7778, \quad E[Y \mid X = x_5] = 8.2727$$